src/query/query_hash_scan.c

```diff
@@ -1924,7 +1924,8 @@ fhs_binary_search_bucket (THREAD_ENTRY * thread_p, PAGE_PTR bucket_page_p, PGSLO
 	}
       else
 	{
-	  low = (flag > 0) ? middle + flag : middle + 1;
+	  middle += (flag > 0) ? flag - 1 : 0; /* go to last slot using flag */
+	  low = middle + 1;
```

Comment:
The variable low will have the same value after this change as before regardless of flag > 0 or not because the earlier code is run as "low = (flag > 0) ? (middle + flag) : (middle + 1);"
Then, is it the real point of this change to increase middle by (flag - 1) when (flag > 0) ?

Reply:
It looks like the logic to set the **middle** value, keeping the **low** value as before.
If that's the intention, it's the right code.

Reply:
The value of middle becomes the final return value. The problem occurred because middle is not changed.

Reply:
OK. I got it.

Reply:
If (compare_result < 0) does not need to adjust the middle value?

Reply:
The flag only can be used when moving to the end.